### ISSUE TITLE
Plugin don't have elementary type checking

### ISSUE THREAD

**Issue by Samehadar** [intellij-rust/intellij-rust#1251]
I believe that the plugin should display warnings in similar situations:
````
fn main() {
    //first example
    let res: i32 = LFSR_Fibonacci;
    res();

    //second example
    len x: i32 = "str";
}

fn LFSR_Fibonacci() {
    println!("Some code");
}

**Comment by matklad** [comment]
Yeah, the type inference is very incomplete at the moment :( 

The problem with showing errors with partially complete type inference is that there's a high chance of false positives.

**Comment by vlad20012** [comment]
Can we assume that Cargo check annotator fixes it?
